gemhardware/devices/src/common/amc/HwGenericAMC.cc

```diff
@@ -293,312 +218,249 @@ bool gem::hw::HwGenericAMC::linkCheck(uint8_t const& gtx, std::string const& opM
 {
   try {
-    req = wisc::RPCMsg("amc.configureDAQModule");
-    req.set_word("enableZS",     enableZS);
-    req.set_word("doPhaseShift", doPhaseShift);
-    req.set_word("runType",      runType);
-    req.set_word("marker",       marker);
-    req.set_word("relock",       relock);
-    req.set_word("bc0LockPSMode",bc0LockPSMode);
-    try {
-      rsp = rpc.call_method(req);
-      try {
-        if (rsp.get_key_exists("error")) {
-          std::stringstream errmsg;
-          errmsg << rsp.get_string("error");
-          CMSGEMOS_ERROR("HwGenericAMC::configureDAQModule: " << errmsg.str());
-          XCEPT_RAISE(gem::hw::devices::exception::RPCMethodError, errmsg.str());
-        }
-      } STANDARD_CATCH;
-    } STANDARD_CATCH;
-  } GEM_CATCH_RPC_ERROR("HwGenericAMC::enableDAQLink", gem::hw::devices::exception::Exception);
+    xhal::common::rpc::call<::amc::daq::configureDAQModule>(rpc, enableZS, runType, doPhaseShift, relock, bc0LockPSMode);
+  } GEM_CATCH_RPC_ERROR("HwGenericAMC::configureDAQModule", gem::hw::devices::exception::Exception);
 }
 
-void gem::hw::HwGenericAMC::enableDAQLink(uint32_t const& enableMask)
+void gem::hw::amc::HwGenericAMC::enableDAQLink(uint32_t const& enableMask)
+//FIXME the meaning of the remote method has changed!!!
 {
   try {
-    req = wisc::RPCMsg("amc.enableDAQLink");
-    req.set_word("enableMask",enableMask);
-    try {
-      rsp = rpc.call_method(req);
-      try {
-        if (rsp.get_key_exists("error")) {
-          std::stringstream errmsg;
-          errmsg << rsp.get_string("error");
-          CMSGEMOS_ERROR("HwGenericAMC::enableDAQLink: " << errmsg.str());
-          XCEPT_RAISE(gem::hw::devices::exception::RPCMethodError, errmsg.str());
-        }
-      } STANDARD_CATCH;
-    } STANDARD_CATCH;
+    xhal::common::rpc::call<::amc::daq::enableDAQLink>(rpc, enableMask);
   } GEM_CATCH_RPC_ERROR("HwGenericAMC::enableDAQLink", gem::hw::devices::exception::Exception);
 }
 
-void gem::hw::HwGenericAMC::disableDAQLink()
+void gem::hw::amc::HwGenericAMC::disableDAQLink()
 {
   try {
-    req = wisc::RPCMsg("amc.disableDAQLink");
-    try {
-      rsp = rpc.call_method(req);
-      try {
-        if (rsp.get_key_exists("error")) {
-          std::stringstream errmsg;
-          errmsg << rsp.get_string("error");
-          CMSGEMOS_ERROR("HwGenericAMC::disableDAQLink: " << errmsg.str());
-          XCEPT_RAISE(gem::hw::devices::exception::RPCMethodError, errmsg.str());
-        }
-      } STANDARD_CATCH;
-    } STANDARD_CATCH;
-  } GEM_CATCH_RPC_ERROR("HwGenericAMC::enableDAQLink", gem::hw::devices::exception::Exception);
+    xhal::common::rpc::call<::amc::daq::disableDAQLink>(rpc);
+  } GEM_CATCH_RPC_ERROR("HwGenericAMC::disableDAQLink", gem::hw::devices::exception::Exception);
 }
 
-void gem::hw::HwGenericAMC::setZS(bool en)
+void gem::hw::amc::HwGenericAMC::setZS(bool en)
```

Comment:
don't all these functions following have `ctp7_modules` equivalents?
why only half update the codebase?
or is this part of the 
> * lots of obsolete functions which require interface redesign

Reply:
yes